x-pack/plugin/sql/qa/jdbc/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/PreparedStatementTestCase.java

```diff
@@ -137,6 +153,61 @@ public void testDatetime() throws IOException, SQLException {
         }
     }
 
+    public void testDatetime_DateNanos() throws IOException, SQLException {
+        assumeTrue("Driver version [" + JDBC_DRIVER_VERSION + "] doesn't support DATETIME with nanosecond resolution]",
+                versionSupportsDateNanos());
+
+        long randomTimestampWitnNanos = randomNanos();
+        int randomNanosOnly = extractNanosOnly(randomTimestampWitnNanos);
+        setupIndexForDateTimeTests(randomTimestampWitnNanos, true);
+
+        try (Connection connection = esJdbc()) {
+            try (PreparedStatement statement = connection.prepareStatement(
+                    "SELECT id, test_date_nanos FROM emps WHERE test_date_nanos = ?")) {
+                Timestamp ts = new Timestamp(toMilliSeconds(randomTimestampWitnNanos));
+                statement.setObject(1, ts);
+                try (ResultSet results = statement.executeQuery()) {
+                    assertFalse(results.next());
+                }
+
+                ts.setNanos(randomNanosOnly);
+                statement.setObject(1, ts);
+                try (ResultSet results = statement.executeQuery()) {
+                    assertTrue(results.next());
+                    assertEquals(1002, results.getInt(1));
+                    assertEquals(ts, results.getTimestamp(2));
+                    assertFalse(results.next());
+                }
+            }
+        }
+    }
+
+    public void testDatetime_DateNanos_OldDrivers() throws IOException, SQLException {
```

Comment:
I chose to use the underscores there so that those methods with the `assumeTrue/assumeFalse` are visually spotted easier, but I don't insist on this, if you prefer the traditional naming will go for it.